code/modules/busy_space/air_traffic.dm

```diff
@@ -0,0 +1,128 @@
+	var/next_message					//When the next message should happen in world.time
+	var/force_chatter_type				//Force a specific type of messages
+
+	var/squelched = 0					//If ATC is squelched currently
+
+/datum/lore/atc_controller/New()
+	spawn(30 SECONDS) //Lots of lag at the start of a shift.
+		msg("New shift beginning, resuming traffic control.")
+	next_message = world.time + rand(delay_min, delay_max)
+	process()
+
+/datum/lore/atc_controller/proc/process()
+	if(world.time >= next_message)
+		if(squelched)
+			next_message = world.time + backoff_delay
+		else
+			next_message = world.time + rand(delay_min,delay_max)
+			random_convo()
+
+	spawn(1 MINUTES) //We don't really need high-accuracy here.
+		process()
+
+/datum/lore/atc_controller/proc/msg(var/message,var/sender)
+	ASSERT(message)
+	global_announcer.autosay("[message]", sender ? sender : "[using_map.station_short] Space Control")
+
+/datum/lore/atc_controller/proc/reroute_traffic(var/yes = 1)
+	if(yes)
+		if(!squelched)
+			msg("Rerouting traffic away from [using_map.station_name].")
+		squelched = 1
+	else
+		if(squelched)
+			msg("Resuming normal traffic routing around [using_map.station_name].")
+		squelched = 0
+
+/datum/lore/atc_controller/proc/shift_ending(var/evac = 0)
+	msg("Automated Shuttle departing [using_map.station_name] for [using_map.dock_name] on routine transfer route.", "NT Automated Shuttle")
+	sleep(5 SECONDS)
+	msg("Automated Shuttle, cleared to complete routine transfer from [using_map.station_name] to [using_map.dock_name].")
+
+/datum/lore/atc_controller/proc/random_convo()
+	var/one = pick(loremaster.organizations) //These will pick an index, not an instance
+	var/two = pick(loremaster.organizations)
+
+	var/datum/lore/organization/source = loremaster.organizations[one] //Resolve to the instances
+	var/datum/lore/organization/dest = loremaster.organizations[two]
+
+	//Let's get some mission parameters
+	var/owner = source.short_name					//Use the short name
+	var/prefix = pick(source.ship_prefixes)			//Pick a random prefix
+	var/mission = source.ship_prefixes[prefix]		//The value of the prefix is the mission type that prefix does
+	var/shipname = pick(source.ship_names)			//Pick a random ship name to go with it
+	var/destname = pick(dest.destination_names)			//Pick a random holding from the destination
+
+	var/combined_name = "[owner] [prefix] [shipname]"
+	var/alt_atc_names = list("[using_map.station_short] TraCon","[using_map.station_short] Control","[using_map.station_short] STC","[using_map.station_short] Airspace")
+	var/wrong_atc_names = list("Sol Command","Orion Control", "[using_map.dock_name]")
+	var/mission_noun = list("flight","mission","route")
+	var/request_verb = list("requesting","calling for","asking for")
+
+	//First response is 'yes', second is 'no'
+	var/requests = list("[using_map.station_short] transit clearance" = list("permission for transit granted", "permission for transit denied, contact regional on 953.5"),
```

Comment:
i'm going to throw out some more correct atc phraseology just for the immulshions, what is a good radio chatter without a good roger and wilco?

"cleared to transit", "unable to approve, contact regional"